guards/core/guard_manager.py

```diff
@@ -36,15 +41,19 @@ def __run_guard(self, guard, event, *args, **kwargs):
             raise GuardException(guard.get_effective_guard())
 
     def run(self, event: Any = RUN_EVENT, *args, **kwargs):
-        closed_guards = []
-        guards = self.listeners.get(event, [])
-        for guard in guards:
-            self.__run_guard(guard, event, *args, **kwargs)
-            if guard.closed:
-                closed_guards.append(guard)
-
-        for closed_guard in closed_guards:
-            guards.remove(closed_guard)
+        self.__running = True
```

Comment:
If we have 1GuardManager in 1GuardContext and this is running on a different thread, maybe we should have an `if __running: return` condition in before_run so it can work like a mutex

Reply:
It is not intended to run on multiple threads for many reasons, so I didn't bother.